EIPS/eip-1895.md

```diff
@@ -0,0 +1,158 @@
+---
+eip: 1895
+title: Support for an Elliptic Curve Cycles
+author: Alexandre Belling <alexandrebelling8@gmail.com>
+discussions-to: https://ethresear.ch/t/reducing-the-verification-cost-of-a-snark-through-hierarchical-aggregation/5128/7
+status: Draft
+type: Standards Track
+category: Core
+created: 2018-31-03
+---
+
+# Support for an Elliptic Curve Cycles
+
+## Simple Summary
+
+The EVM currently supports elliptic curves operations for curve *alt-bn128* thanks to precompiles `ecadd` and `ecmul` and `ecpairing`. The classes MNT4 and 6 contains cycles of curves and this enable doing operation on one curve inside a SNARK on the other (end reversely). This EIP suggests adding support for those curves.
+
+## Abstract
+
+Adds supports for the following operations through precompiles:
+
+* `ecadd` on MNT4
+* `ecmul` on MNT4
+* `ecpairing` on MNT4
+
+## Motivation
+
+Elliptic curve is the basic block of recursive SNARKSs (ie: verifying a SNARK inside a SNARK) and this addresses the issue of scalable zero-knowledge. More generally this addresses partly the scalability issue as SNARKs verification are constant time in the size of the circuit being verified.
```

Comment:
'SNARKSs' -> 'SNARKs'

I also don't fully understand the last sentence here. 

_"More generally this addresses partly the scalability issue as SNARKs verification are constant time in the size of the circuit being verified."_